File: find_bus.js
const fs = require('fs');

const buf = fs.readFileSync('college.glb');
const magic = buf.toString('utf8', 0, 4);
if (magic !== 'glTF') {
  console.log('Not a glb');
  process.exit(1);
}

const version = buf.readUInt32LE(4);
const length = buf.readUInt32LE(8);
const chunk0Length = buf.readUInt32LE(12);
const chunk0Type = buf.toString('utf8', 16, 20);

if (chunk0Type !== 'JSON') {
  console.log('Chunk 0 not JSON');
  process.exit(1);
}

const jsonData = buf.toString('utf8', 20, 20 + chunk0Length);
const gltf = JSON.parse(jsonData);

let count = 0;
gltf.nodes.forEach((node, i) => {
  const name = (node.name || '').toLowerCase();
  if (name.includes('school_bus')) {
    console.log(`Node ${i}: ${name}`);
    if (node.translation) console.log(`  Translation: ${node.translation}`);
    count++;
  }
});
console.log(`Total: ${count}`);

File: find_bus.py
import json
import struct

def parse_glb(file_path):
    with open(file_path, 'rb') as f:
        magic = f.read(4)
        if magic != b'glTF':
            print("Not a GLB file")
            return
        
        version = struct.unpack('<I', f.read(4))[0]
        length = struct.unpack('<I', f.read(4))[0]
        
        chunk0_length = struct.unpack('<I', f.read(4))[0]
        chunk0_type = f.read(4)
        if chunk0_type != b'JSON':
            print("Chunk 0 is not JSON")
            return
            
        json_data = f.read(chunk0_length).decode('utf-8')
        gltf = json.loads(json_data)
        
        nodes = gltf.get('nodes', [])
        for i, node in enumerate(nodes):
            name = node.get('name', '').lower()
            if 'school_bus' in name:
                # get translation if any
                translation = node.get('translation', [0, 0, 0])
                print(f"Node {i}: {name}, Position: {translation}")

parse_glb('college.glb')
